src/nvim/terminal.c

```diff
@@ -241,6 +243,7 @@ Terminal *terminal_open(TerminalOptions opts)
   set_option_value((uint8_t *)"wrap", false, NULL, OPT_LOCAL);
   set_option_value((uint8_t *)"number", false, NULL, OPT_LOCAL);
   set_option_value((uint8_t *)"relativenumber", false, NULL, OPT_LOCAL);
+  buf_set_term_title(curbuf, cstr_to_string((char *)curbuf->b_ffname));
```

Comment:
This is a memory leak, because `buf_set_term_title()` does not take ownership of the String argument.